secant/Features/App/AppStore.swift

```diff
@@ -192,12 +195,26 @@ extension AppReducer {
                     birthday: birthday,
                     with: environment
                 )
-                try environment.SDKSynchronizer.prepareWith(initializer: initializer)
-                try environment.SDKSynchronizer.start()
+
+                return Effect.task {
+                    do {
+                        try await environment.SDKSynchronizer.prepareWith(initializer: initializer, seedBytes: seedBytes)
+                        try environment.SDKSynchronizer.start()
+                        return .sdkInitialized
+                    } catch {
+                        return .sdkInitializationFailed
+                    }
+                }
             } catch {
-                state.appInitializationState = .failed
-                // TODO [#221]: error we need to handle (https://github.com/zcash/secant-ios-wallet/issues/221)
+                return Effect(value: .sdkInitializationFailed)
             }
+
+        case .sdkInitialized:
+            return .none
+
+        case .sdkInitializationFailed:
+            // TODO [#221]: error we need to handle (https://github.com/zcash/secant-ios-wallet/issues/221)
+            state.appInitializationState = .failed
             return .none
```

Comment:
After one more deep look and thinking about it, I would lean to refactor SDKSynchronizer's `prepareWith()` to be non-async. The only reason why it's async is the call of blockProcessor's `setStartHeight`. That is complicating a lot of stuff on the SDKSynchronizer's side and it's very similar like we had start() stop() async and purposely refactored to synced APIs. I believe that `setStartHeight` can be encapsulated into Task just like start and stop are.

@pacu @Chlup, what do you think about this proposal? In general at this point we probably want SDSycnhronizer's as much non-async as possible.

Reply:
The problem with that is that prepare(with:) performs migrations and could be slow. But I agree that having an async API on the SDKSynchronizer at the moment is a pain.  Should have both alternatives.